### ISSUE TITLE
Pydanticgen produces `ConfiguredBaseModel` with unexpected keywords

### ISSUE THREAD

**Issue by caufieldjh** [linkml/linkml#1632]
**Describe the bug**
Pydanticgen appears to produce a `ConfiguredBaseModel` with unexpected keyword arguments - at least per Pydantic v.2.2+.

It generates a class like this:
```
class ConfiguredBaseModel(BaseModel,
                validate_assignment = True,
                validate_default = True,
                extra = 'forbid',
                arbitrary_types_allowed = True,
                use_enum_values = True):
    pass
```
or like the one here: https://github.com/linkml/linkml/blob/main/tests/test_data/model/personinfo_pydantic.py#L18-L26

This still works without issue on Pydantic 2, but BaseModel also expects those configuration settings like `extra` to be set a bit differently (a la https://docs.pydantic.dev/2.2/usage/models/#extra-fields).

So this is only really an issue because:
* The class may not be initialized with those extra params
* This throws errors with Mypy, like
```
error: Unexpected keyword argument "extra" for "__init_subclass__" of "object"  [call-arg]
```

**Comment by cmungall** [comment]
#1641

**Comment by kevinschaper** [comment]
I believe this is fixed in #1660, which now defines the configuration with: 
```
    model_config = ConfigDict(
        validate_assignment=True,
        validate_default=True,
        extra='forbid',
        arbitrary_types_allowed=True,
        use_enum_values = True)
```
